client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,164 @@
+  const [dateRange, setDateRange] = useState<Record<string, Date | null>>({ from: null, to: null });
+  const [formattedDateRange, setFormattedDateRange] = useState<string>('any');
+  const [displayedUsers, setDisplayedUsers] = useState<User[]>(users.slice(0, 6));
+  const [snackbarOpen, setSnackbarOpen] = useState<true | false>(false);
+  const [search, setSearch] = useState<string>('');
+
+  const moment = extendMoment(Moment);
+
+  const handleShowMore = () => {
+    const numberOfUsers = displayedUsers.length;
+    const lastUser = users[displayedUsers.length];
+    if (lastUser === undefined) {
+      setSnackbarOpen(true);
+    }
+    setDisplayedUsers(users.slice(0, numberOfUsers + 3));
+  };
+
+  const handleSnackbarClose = (event?: React.SyntheticEvent, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
+
+  const handleCalendarOpen = () => {
+    setCalendarOpen((prevState) => !prevState);
+  };
+
+  const updateDateRange = (dateFrom: Date | null, dateTo: Date | null) => {
+    if (dateFrom instanceof Date && dateTo instanceof Date) {
+      setDateRange({
+        from: dateFrom,
+        to: dateTo,
+      });
+      const formattedDateFrom = moment(dateFrom).format('MMMM Do YYYY');
+      const formattedDateTo = moment(dateTo).format('MMMM Do YYYY');
+      setFormattedDateRange(`${formattedDateFrom} – ${formattedDateTo}`);
+    } else {
+      setFormattedDateRange(`any`);
+    }
+  };
+
+  const updateSearch = () => {
+    const searchResults: User[] = [];
+    setDisplayedUsers([]);
+    users.map((user: User) => {
+      if (user.city.toLowerCase().includes(search)) {
+        searchResults.push(user);
+        setDisplayedUsers(searchResults);
+      }
+    });
+  };
+
+  //   Updates based on date selection
+  useEffect(() => {
+    if (formattedDateRange === 'any') {
+      setDisplayedUsers(users.slice(0, 6));
+    } else {
+      if (dateRange.from && dateRange.to) {
+        const searchResults: User[] = [];
+        setDisplayedUsers([]);
+        const range = moment.range(dateRange.from, dateRange.to);
+        users.map((user) => {
+          //   If at least one available date falls within the selected range, the user
+          //   is added to the displayedUsers array.
+          const isInDateRange = (date: Date) => {
+            return range.contains(date);
+          };
+          if (user.availableDates.some(isInDateRange)) {
+            searchResults.push(user);
+            setDisplayedUsers(searchResults);
+          }
+        });
+      }
+    }
+  }, [formattedDateRange, setFormattedDateRange]);
+
+  //   Updates based on search parameters
+  useEffect(() => {
+    if (search) {
+      updateSearch();
+    } else {
+      setDisplayedUsers(users.slice(0, 6));
+    }
+  }, [search, setSearch]);
+
+  const profileCardGrid = (
+    <Grid container className={classes.profilesContainer} xs={9}>
```

Comment:
You should only apply breakpoints to grid items [https://material-ui.com/api/grid/](https://material-ui.com/api/grid/)

Reply:
This is rendered as a grid item in the return statement of ProfileListings()